### PR TITLE
Fix various pfcwd config_db check failure

### PR DESCRIPTION
**Pull request by auspham**
### Description of PR

Summary:
Fixes # (issue) 29826096, 29826079, 29808432

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

### Back port request
- [ ] 202012
- [ ] 202205
- [ ] 202305
- [ ] 202311
- [x] 202405

#### What is the motivation for this PR?
Currently, for PFCWD, we are setting the config_db a certain way for our test case. However, after the tear down, we do not recover the ConfigDB setting to the original which leads to config_db_check issue

#### How did you do it?

This PR attempt to fix 
- test_pfcwd_all_port_storm
- test_pfcwd_timer_accuracy
- pfcwd/test_pfc_config

Since they're similar

#### How did you verify/test it?
I've verified on T2 chassis switch

### CODE REVIEW

File: tests/pfcwd/test_pfc_config.py [hunk 1]
```diff
@@ -158,11 +158,15 @@ def stop_pfcwd(duthosts, enum_rand_one_per_hwsku_frontend_hostname):
...
     """
-    yield
     duthost = duthosts[enum_rand_one_per_hwsku_frontend_hostname]
     logger.info("--- Stop Pfcwd --")
     duthost.command("pfcwd stop")
 
+    yield
+
+    logger.info("--- Start Pfcwd--")
+    duthost.command("pfcwd start_default")
```
Comment (auspham) [inline tests/pfcwd/test_pfc_config.py:168]: Hi @ZhaohuiS, I've seen your concern here for this https://github.com/sonic-net/sonic-mgmt/pull/5392#discussion_r884463037

The problem with putting `yield` first is that it doesn't reset pfcwd to the default setting as we should be doing after this module

File: tests/pfcwd/test_pfc_config.py [hunk 2]
```diff
@@ -320,7 +324,7 @@ def test_default_cfg_after_load_mg(self, duthosts, enum_rand_one_per_hwsku_front
             None
         """
         duthost = duthosts[enum_rand_one_per_hwsku_frontend_hostname]
-        config_reload(duthost, config_source='minigraph')
+        config_reload(duthost, config_source='minigraph', safe_reload=True)
```
Comment (auspham) [inline tests/pfcwd/test_pfc_config.py:327]: Also fyi @ZhaohuiS in here since we're doing a minigraph config_reload. It's unavoidable that we're going to experience config_db_check fail. I've added a safe_reload=True here as suggested by @yejianquan to keep the testbed stable

File: tests/pfcwd/test_pfcwd_timer_accuracy.py [hunk 3]
```diff
@@ -64,7 +69,7 @@ def ignore_loganalyzer_exceptions(enum_rand_one_per_hwsku_frontend_hostname, log
 
 @pytest.fixture(scope='class', autouse=True)
 def pfcwd_timer_setup_restore(setup_pfc_test, enum_fanout_graph_facts, duthosts,        # noqa F811
-                              enum_rand_one_per_hwsku_frontend_hostname, fanouthosts):
+                              enum_rand_one_per_hwsku_frontend_hostname, fanouthosts, stop_pfcwd):
```
Comment (auspham) [inline tests/pfcwd/test_pfcwd_timer_accuracy.py:72]: For this fixture here, I remove the autouse since the behavior is unexected. We want our fixture to be this following order:

1. stop_pfcwd
2. pfcwd_timer_setup_restore
3. pfcwd_timer_setup_restore clean up
4. stop_pfcwd clean up

This change here will make sure that this happen.

File: tests/pfcwd/test_pfc_config.py (hunk 1 above)
Comment (augusdn) [reply tests/pfcwd/test_pfc_config.py:168]: This fixture seems identical to the one in tests/pfcwd/test_pfcwd_all_port_storm.py
1. can we combine these two and put it into helper file?
2. I see this one is scoped at "function" level and other at "class" level. What would be the reason we have two different scopes?
3. They both have autouse=True, which will then called for other tests automatically. Do we want to call this fixture for other tests as well?

File: tests/pfcwd/test_pfcwd_all_port_storm.py [hunk 4]
```diff
@@ -85,6 +85,11 @@ def stop_pfcwd(duthosts, enum_rand_one_per_hwsku_frontend_hostname):
     logger.info("--- Stop Pfcwd --")
     duthost.command("pfcwd stop")
 
+    yield
+
+    logger.info("--- Start Pfcwd --")
+    duthost.command("pfcwd start_default")
```
Comment (augusdn) [inline tests/pfcwd/test_pfcwd_all_port_storm.py:91]: commented above

File: tests/pfcwd/test_pfcwd_timer_accuracy.py [hunk 5]
```diff
@@ -40,6 +40,11 @@ def stop_pfcwd(duthosts, enum_rand_one_per_hwsku_frontend_hostname):
     logger.info("--- Stop Pfcwd --")
     duthost.command("pfcwd stop")
 
+    yield
+
+    logger.info("--- Start Pfcwd --")
```
Comment (augusdn) [inline tests/pfcwd/test_pfcwd_timer_accuracy.py:45]: Same as above, I think we can combine into one file. Also wondering why we have two different scopes

File: tests/pfcwd/test_pfc_config.py (hunk 1 above)
Comment (augusdn) [reply tests/pfcwd/test_pfc_config.py:168]: maybe in pfcwd_helper.py

Comment (auspham) [reply tests/pfcwd/test_pfc_config.py:168]: The scope "function" and "class" were originally there and it passes our nightly so i didn't spend time looking into it. The reason why for `all_port_storm` we have class is that we only have 1 test in 1 class.

In this module we want to use before every tests but for the other module we don't apply the same

File: tests/pfcwd/test_pfc_config.py [hunk 6]
```diff
@@ -158,11 +158,15 @@ def stop_pfcwd(duthosts, enum_rand_one_per_hwsku_frontend_hostname):
     Returns:
         None
     """
-    yield
     duthost = duthosts[enum_rand_one_per_hwsku_frontend_hostname]
     logger.info("--- Stop Pfcwd --")
     duthost.command("pfcwd stop")
 
+    yield
```
Comment (ZhaohuiS) [inline tests/pfcwd/test_pfc_config.py:165]: @auspham does it revert the change in https://github.com/sonic-net/sonic-mgmt/pull/5392/files#diff-2b591708001ec52b172b72036e80a4c635a8cc8476747d95d6fa8f59b0ca3981?

@lipxu  @antonptashnik could you please review this change?

Comment (auspham) [reply tests/pfcwd/test_pfc_config.py:165]: @ZhaohuiS yes, the problem with putting `yield` first is that it doesn't reset pfcwd to the default setting as we should be doing after this module

File: tests/pfcwd/test_pfcwd_timer_accuracy.py [hunk 7]
```diff
@@ -40,6 +40,11 @@ def stop_pfcwd(duthosts, enum_rand_one_per_hwsku_frontend_hostname):
     logger.info("--- Stop Pfcwd --")
     duthost.command("pfcwd stop")
 
+    yield
```
Comment (lipxu) [inline tests/pfcwd/test_pfcwd_timer_accuracy.py:43]: If the stop/start apply to all the cases, could we include it in the fixture at the module level? thanks

**Review by lipxu — APPROVED** [review]

File: tests/pfcwd/test_pfcwd_timer_accuracy.py (hunk 7 above)
Comment (auspham) [reply tests/pfcwd/test_pfcwd_timer_accuracy.py:43]: adjusted the scope to module @lipxu
